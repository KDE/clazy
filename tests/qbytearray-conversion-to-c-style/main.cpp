#include <QtCore/QByteArray>
#include <QtCore/QString>

static void func(const char *str) { }

static QByteArray returnBa() { return "foo"; }

static bool boolFunc(const char *s) { return s != nullptr; }

#define TEXT "macrotext"

struct S {
    QString key;
    QByteArray keyAsUtf8() { return key.toUtf8(); }
};

void test()
{
    QByteArray ba = "this is a qbytearray";

    const char *c = ba;
    func(ba);

    func(returnBa());

    func(QByteArray("foo"));
    func(QByteArray(TEXT " ba"));
    func(QByteArrayLiteral("ba-literal"));
    func(QByteArrayLiteral(TEXT " ba-literal"));

    QString utf16 = QString::fromLatin1("string");
    func(std::exchange(utf16, {}).toLocal8Bit());

    if (boolFunc(returnBa())) {
    }

    if (__builtin_expect(!!(boolFunc(returnBa())), true)) {
    }
    if (Q_LIKELY(boolFunc(returnBa()))) {
    }

    S s{};
    func(s.key.toUtf8());
    func(s.keyAsUtf8());

    func("foo" + ba + "bar");
}
