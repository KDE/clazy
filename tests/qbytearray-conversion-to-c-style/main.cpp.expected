qbytearray-conversion-to-c-style/main.cpp:21:21: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:22:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:24:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:26:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:27:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:28:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:29:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:32:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:34:18: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:37:38: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:39:27: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:43:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:44:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
qbytearray-conversion-to-c-style/main.cpp:46:10: warning: Don't rely on the QByteArray implicit conversion to 'const char *'. [-Wclazy-qbytearray-conversion-to-c-style]
